libcore/src/events/EventManager.cpp

```diff
@@ -38,74 +38,42 @@ EventManager::EventManager(Building * _b)
     _building = _b;
 }
 
-void EventManager::AddEvents(std::vector<Event> events)
+void EventManager::AddEvent(std::unique_ptr<Event> event)
 {
-    _events.insert(_events.end(), events.begin(), events.end());
+    event->SetBuilding(_building);
+    _events.emplace_back(std::move(event));
+    std::sort(std::begin(_events), std::end(_events), [](auto & event1, auto & event2) {
```

Comment:
Couldn't we use a sorted data structure and insert the element at the correct position instead of resorting every time?

Reply:
After discussion: no changes. At the moment is the number of events not significant to raise performance issues.